### ISSUE TITLE
longer response time of the first request than expected

### ISSUE THREAD

**Issue by muhaook** [open-telemetry/opentelemetry-dotnet-instrumentation#2307]
the sample application was running on IIS. env info:
- Windows Server 2022 
- CPU: 1 Core, 2 logical processors, base speed 2.19GHz
- Memory: 12GB

The sample I was using is a pretty simple WebApiCore helloworld. but it still took 4-5 sec to get reponse from the very first request. It is likely because of the poor CPU on my VM.

after install otel-auto-instrumentation with enabled bytecode instrumentation and source instrumentation, I found the first request became slower. it took about 14 sec to get the response.

When I disabled bytecode instrumentation but enabled .NET source instrumentation using DOTNET_STARTUP_HOOKS, the response time was about 7 sec.

Below is what I did to disable bytecode instrumentation but enable .NET source instrumentation:
```
#COR_ENABLE_PROFILING=1
#CORECLR_ENABLE_PROFILING=1
DOTNET_STARTUP_HOOKS=c:\tracer-home\net\OpenTelemetry.AutoInstrumentation.StartupHook.dll
```

Here are details:
### all enabled on otel-auto-instrumentation:
![enableAll](https://user-images.githubusercontent.com/52417764/223781531-31164199-8043-49a6-9801-b096e048569a.png)
logs: 
[logs_enableAll.zip](https://github.com/open-telemetry/opentelemetry-dotnet-instrumentation/files/10923121/logs_enableAll.zip)

### enabled source instrumentation only:
![enableSourceInstumentatOnly](https://user-images.githubusercontent.com/52417764/223782039-fb1b793c-2253-402f-ba71-09d7e44def30.png)
logs: 
[logs_enableSourceInstrumentOnly.zip](https://github.com/open-telemetry/opentelemetry-dotnet-instrumentation/files/10923127/logs_enableSourceInstrumentOnly.zip)

### disabled otel-auto-instrumentation
![disabled](https://user-images.githubusercontent.com/52417764/223782168-11481bb4-1662-491a-98d4-3d8e052bd351.png)

The sample I was using: 
[WebApiStartup.zip](https://github.com/open-telemetry/opentelemetry-dotnet-instrumentation/files/10923114/WebApiStartup.zip)

**Comment by rajkumar-rangaraj** [comment]
I used the same app to test the behavior and could not reproduce it. For StartupHook Auto-Instrumentation it takes ~600 to ~700 ms which is acceptable.

```
Without Instrumentation
2023-03-29 18:48:25 ::1 GET /api/hello - 8091 - ::1 Mozilla/5.0+(Windows+NT+10.0;+Win64;+x64)+AppleWebKit/537.36+(KHTML,+like+Gecko)+Chrome/111.0.0.0+Safari/537.36 - 200 0 0 693

Auto-Instrumentation with StartupHook
2023-03-29 18:39:25 ::1 GET /api/hello - 8091 - ::1 Mozilla/5.0+(Windows+NT+10.0;+Win64;+x64)+AppleWebKit/537.36+(KHTML,+like+Gecko)+Chrome/111.0.0.0+Safari/537.36 - 200 0 0 1259

Auto-Instrumentation with CLR Profiler
2023-03-29 18:45:07 ::1 GET /api/hello - 8091 - ::1 Mozilla/5.0+(Windows+NT+10.0;+Win64;+x64)+AppleWebKit/537.36+(KHTML,+like+Gecko)+Chrome/111.0.0.0+Safari/537.36 - 200 0 0 2202
```

**Comment by Kielek** [comment]
Removing 0.7.0 milestone to revisit on the next SIG meeting how to handle this case. @rajkumar-rangaraj, do you have any recommendation?

**Comment by muhaook** [comment]
@Kielek , i do not think this is a big issue. it is unlikely that PROD env has only 1 CPU. please feel free to postpone or close this issue.
